src/query/catalog/src/table_context.rs

```diff
@@ -101,6 +101,9 @@ pub trait TableContext: Send + Sync {
     fn get_cacheable(&self) -> bool;
     fn set_cacheable(&self, cacheable: bool);
 
+    fn set_snapshot(&self, location: String, ver: u64);
+    fn get_snapshot(&self) -> Option<(String, u64)>;
```

Comment:
I think type `Location` is better to represent `location: String, ver: u64` tuple